base/repl/LineEdit.jl

```diff
@@ -572,15 +592,18 @@ end
 edit_clear(buf::IOBuffer) = truncate(buf, 0)
 
 function edit_clear(s::MIState)
+    push_undo(s)
```

Comment:
maybe you could `push_undo` only if the buffer is not empty already? Another possibility could also be to not push a new state in `push_undo` when it's equal to the last pushed state?